contracts-sdk/src/lib/signTxAuthDataViem.ts

```diff
@@ -91,12 +92,23 @@ export const signTxAuthDataLib = async (
   const functionCallData = generateFunctionCallDataViem(
     txAuthInput.contractAbi as unknown as Abi,
     txAuthInput.functionName,
-    [...txAuthInput.args, blockExpiration, "0x1234"]
+    [...txAuthInput.args]
   );
 
-  // remove 64 bytes (32 bytes for the length and 32 bytes for the fake signature itself)
-  // = 128 characters
-  const argsWithSelector = functionCallData.slice(0, -128) as Address;
+  const length = ethers.utils.hexZeroPad(
+    ethers.BigNumber.from(96).toHexString(),
+    32
+  );
+
+  const abiEncodedBlockExpiration = ethers.utils.hexZeroPad(
+    ethers.BigNumber.from(blockExpiration).toHexString(),
+    32
+  );
+
+  // Remove the placeholder for the signature
+  const argsWithSelector = (functionCallData +
```

Comment:
I don't really understand why block Exploration needs to be part of argsWithSelector if it is not in the args in the ABI

Reply:
You're right, now the offset was fixed to get the correct one in the contract and the sdk/tests were changed too